docs/generator.md

```diff
@@ -1,16 +1,265 @@
+author: fetchai
+version: 0.1.0
+license: Apache 2.0
+description: 'A protocol for negotiation over a fixed set of resources involving two parties.'
+speech_acts:
+  cfp:
+    query: ct:DataModel
+  propose:
+    query: ct:DataModel
+    price: pt:float
+  accept: {}
+  decline: {}
+  match_accept: {}
+```
+
+
+Each protocol specification yaml file contains some basic information about the protocol. The allowed fields and what they represent are:
+
+ * `name`: The name of the protocol (written in <a href="https://en.wikipedia.org/wiki/Snake_case" target="_blank">snake_case</a>)
+ * `authors`: List of authors
+ * `version`: The current version of the protocol
+ * `license`: Licensing information
+ * `description`: A short description of the protocol
+
+Each field is a key/value pair, where both the key and the value are yaml strings. Every field is mandatory. 
+
+Furthermore, the protocol specification must describe the syntax of valid messages according to this protocol.
+Therefore, there is another mandatory field: `speech-acts`, which defines the set of performatives valid under this protocol, and a set of contents (i.e. parameters) for each performative.  
+
+The format of the `speech-act` is as follows:
+`speech-act` is a dictionary, where each key is a unique **performative** (yaml string), and the value is a **content** dictionary. If a performative `perm` does not have any content, then its content dictionary is empty, e.g. simply denoted as `perm: {}`.
+
+Each content dictionary is composed of key/value pairs, where each key is the name of a content (yaml string) and the value is its <a href="../generator/#types">type</a> (yaml string).  
+
+### Types
+
+The specific types which could be assigned to contents in a protocol specification are described in the table below.
+
+Types are either user defined (i.e. custom type) or primitive. 
+
+Custom types are prepended with "ct:" and their format is described using regular expression in the table below. 
+
+Primitive types are prepended with "pt:". 
+
+There are different categories of primitive types, e.g. &lt;PT&gt; such as int and bool, &lt;PCT&gt; such as sets and lists, and so on. 
+
+Primitive types have a compositional format. For example, consider sets under &lt;PCT&gt;, i.e. unordered, immutable collections with no duplicate elements. Sets must describe the type of their elements in square brackets and these could either be &lt;PT&gt; (e.g. `pt:int`, `pt:bool`) or &lt;CT&gt; (i.e. a custom type). Therefore, when describing the format of types in the table below, `/` between two type categories `T1` and `T2` means "or", i.e. "either a `T1` or a `T2` type".
+
+Multi types are an "or" separated set of sub-types. A multi type `pt:int or pt:float` means the type is either `pt:int` or `pt:float`.
+
+An optional type for a content indicates that the content is optional, but if it is present, its type must match what is described in the square brackets. 
+
+<table>
```

Comment:
Please replace this with a simple table as for the cli commands. The one below is very hard to review.

Reply:
Done. 

Although in this new format, I couldn't find a way to make cells which span multiple rows for "Primitive type", "<PT>", "Primitive collection type" and "<PCT>".